packages/server/src/sdk/app/datasources/plus.ts

```diff
@@ -96,6 +97,15 @@ export async function buildSchemaFromSource(
   const datasource = await datasources.get(datasourceId)
 
   const { tables, errors } = await buildFilteredSchema(datasource, tablesFilter)
+
+  const oldTables = datasource.entities || {}
+  const tablesToRemove = Object.keys(oldTables).filter(
+    t => !Object.keys(tables).includes(t)
+  )
+  for (const table of tablesToRemove) {
+    await sdk.rowActions.deleteAll(oldTables[table]._id!)
```

Comment:
I don't like adding the row action stuff mixed with the datasource, but this is the only way to do it without redesigning how this currently works..